ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1512,12 +1411,55 @@ impl<'a> Assemble<'a> {
                                 [RegConstraint::InputIntoReg(op.clone(), r)],
                             );
                         }
-                        _ => todo!(),
+                        reg_alloc::Register::FP(r) => {
+                            let [_] = self.ra.assign_fp_regs(
+                                &mut self.asm,
+                                iidx,
+                                [RegConstraint::InputIntoReg(op.clone(), r)],
+                            );
+                        }
                     }
                 }
-                e => todo!("{:?}", e),
+                _ => todo!(),
             }
         }
+    }
+
+    fn cg_traceloopjump(&mut self) {
+        // Loop the JITted code if the `tloop_start` label is present (not relevant for IR created
+        // by a test or a side-trace).
+        let label = StaticLabel::global("tloop_start");
+        match self.asm.labels().resolve_static(&label) {
+            Ok(_) => {
+                // Found the label, emit a jump to it.
+                self.write_jump_vars(None);
+                dynasm!(self.asm; jmp ->tloop_start);
+            }
+            Err(DynasmError::UnknownLabel(_)) => {
+                // Label not found. This is OK for unit testing, where we sometimes construct
+                // traces that don't loop.
+                #[cfg(test)]
+                {
+                    self.comment(self.asm.offset(), "Unterminated trace".to_owned());
+                    dynasm!(self.asm; ud2);
+                }
+                // FIXME: Side traces don't currently loop back to the parent trace and deopt into
```

Comment:
Oh, I didn't realise this! Should fixing this be the next PR?

Reply:
Err, no that's an old comment...whoops. 😅

Reply:
Ah, OK, I was a bit confused!